File: Suite420-SpringBackEnd/src/main/java/br/edu/ifpb/gugawag/social/model/Produto.java
package br.edu.ifpb.gugawag.social.model;

import javax.persistence.*;

@Entity
@Table(name = "tb_produto")
public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nome;
    private String descricao;
    private Double preco;
    private String imagem;

    public Long getId(){
        return id;
    }
    public String getNome(){
        return nome;
    }
    public void setNome(String nome){
        this.nome = nome;
    }
    public String getDescricao(){
        return descricao;
    }
    public void setDescricao(String descricao){
        this.descricao = descricao;
    }
    public Double getPreco(){
        return preco;
    }
    public void setPreco(Double preco){
        this.preco = preco;
    }
    public String getImagem(){
        return imagem;
    }
    public void setImagem(String imagem){
        this.imagem = imagem;
    }
}

File: Suite420/src/app/shared/model/produto.ts
export class Produto {
  id?: number;
  nome?: string;
  descricao?: string;
  preco?: number;
  imagem?: string;

  constructor(id?: number, produtos: Produto = {}) {
    this.id = id;
    this.nome = produtos.nome;
    this.descricao = produtos.descricao;
    this.preco = produtos.preco;
    this.imagem = produtos.imagem;
  }
}
